src/lib/Permission/PermissionChecker.php

```diff
@@ -225,6 +226,35 @@ public function getContentUpdateLimitations(Location $location): LookupLimitatio
         );
     }
 
+    public function getContentDeleteLimitations(Location $location): LookupLimitationResult
+    {
+        $content = $location->getContent();
+
+        $translations = $content->getVersionInfo()->languageCodes;
```

Comment:
Use of magic detected. Sadly, the current API is not very pleasant for VersionInfo
```suggestion
        $translations = array_map(
            static fn (Language $language): string => $language->getLanguageCode(),
            $content->getVersionInfo()->getLanguages()
        );
```